### PR TITLE
Wyner and exact common informations

### LINKED ISSUE #9: Add the Wyner Common Information
This one is harder. Needs some sort of optimization.

$\min{ I[ Z : {X_i}] : B[{X_i} | Z] = 0 }$

### PR DESCRIPTION
**Pull request by Autoplectic**
Implements both the Wyner and Exact common informations, largely from a common code base.

Some further improvements could certainly be made, for example seeding the optimization with the functional markov chain random variable. Also, the various parameters (temperature, step size) to basinhopping might benefit from tweaking, but doing so requires some knowledge of what the basins look like (approx radius, relative depth).

Overall, this seems to work well.

Closes #9 and #11

### CODE REVIEW

File: dit/multivariate/base_markov_optimizer.py [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+                           accept_test=accept_test,
+                          )
+
+        self._res = res
+
+        success, msg = self._success(res)
+        if not success:
+            raise Exception(msg)
+
+        if polish:
```
Comment (chebee7i) [inline dit/multivariate/base_markov_optimizer.py]: Maybe allow polish to specify the cutoff when it is not False? Or add a cutoff parameter?

**Comment by chebee7i** [comment]
Looks very good. I def still need to read through all this more carefully but was able to reason well about what you were doing. Seems like we've settled on a few coding paradigms.

Going to merge this now. If you want to add the polish comment adter, go ahead.

**Comment by chebee7i** [comment]
Opps, holding off due to travisCI.

**Comment by Autoplectic** [comment]
I'm glad the code was overall pretty readable. I know stuff like the slicing in construct_joint can be hard to wrap one's head around.

**Comment by chebee7i** [comment]
Yeah the ellipsis and colons are always fun (and powerful).

**Pull request merged**